### DISCUSSION TITLE
Is there something wrong with availability reporting

### DISCUSSION THREAD

**Discussion by janhark** [Koenkk/zigbee2mqtt#14443] [General]
I have about 27 devices in zigbee2mqtt, about half are mains-powered (ikea and xiaomi controlled plugs) and half are various battery-powered devices from ikea and xiaomi. In addition, there are 2 ikea signal repeaters. Everything works very well and without problems when it comes to actual functionality of the devices. I have availability turned on with the global setting for all devices with active device timeout being 90 minutes and passive device 60 minutes. I have changed those values multiple times while trying to find a solution to my availability issue.

The house automation is based on OpenHAB, which also controls all of the devices behind zigbee2mqtt and while everything works there when it comes to functionality, the availability reporting seems to be utterly haphazard. At any given time, roughly half of the devices show themselves offline in OpenHAB (while working perfectly regarding switching/sending sensor data) because they have been reported as availability offline at some point. This applies evenly to mains-powered and battery-powered devices, even though all of said devices are actually online 24/7 and constantly communicating. For example the Xiaomi plugs report multiple items probably every 5 seconds or so, but even when monitoring the reporting with MQTT Explorer there might be periods of 10 hours when nothing is reported regarding their availability by zigbee2mqtt. All of the data is passed on by zigbee2mqtt perfectly fine, the only thing not reported is availability.

I just got annoyed at this as I am watching grafana reports from some of the devices regarding location temperatures and power consumption with frequent and stable numbers incoming today, while the exact same devices report themselves as offline in OpenHAB and have received no availability information over the last 24 hours. Even more mysterious, I have altogether 4 Xiaomi temperature and humidity sensors with 1 of them constantly working and reported as online by zigbee2mqtt while the other three are also constantly working but apparently reported as offline by zigbee2mqtt at some point, although not today according to MQTT Explorer. None of the Xiaomi plugs I have have reported anything regarding availability today even though they all send data every 5 seconds. Finally, I have three ikea motion sensors, which are reporting availability very nicely and seemingly correctly.

Have I misunderstood something? Should I configure the availability individually for every device or wherein is the issue? I can see that the availability reporting works partially but can I change the reporting frequency somewhere? I know the timeout configurable is for defining when to consider the device offline but is there a setting somewhere for "report device availability status every X minutes whether online or offline"? Is the reporting maybe device-dependent and ikea/xiaomi happen to be very bad in this sense?

**Comment by NikitaKorneev** [comment]
I just created another discussion that is "adjacent" to this topic. I have a lot of Tuya zigbee spotlights and I don't think they report their state at all. They can only get new state from the command that you send and update themselves but I believe there is no feed back from them.

If I am right, this might be the reason your devices switch to "offline". 

I wish someone could clarify this.

Do your devices have "last seen" attribute? Does it show like a really long time?

**Reply by janhark** [reply]
After further testing I have come to the conclusion the availability reporting works right after a restart of Zigbee2MQTT and max 1-2 days after that. Once this period has passed since last restart, availability reporting for the devices starts being left out one by one until none of them are reporting availability even though they are reporting all data normally.

I did not have "last seen" on - have just turned it on and will observe what happens.
